src/sentry/workflow_engine/endpoints/validators/metric_alert_detector.py

```diff
@@ -1,15 +1,40 @@
+from datetime import timedelta
+from typing import Any, TypedDict
+
 from rest_framework import serializers
 
+from sentry.snuba.models import QuerySubscription, SnubaQuery, SnubaQueryEventType
 from sentry.snuba.snuba_query_validator import SnubaQueryValidator
+from sentry.snuba.subscriptions import update_snuba_query
 from sentry.workflow_engine.endpoints.validators.base import (
     BaseDataConditionGroupValidator,
     BaseDetectorTypeValidator,
     NumericComparisonConditionValidator,
 )
-from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.models import DataConditionGroup, DataSource, Detector
+from sentry.workflow_engine.models.data_condition import Condition, DataCondition
 from sentry.workflow_engine.types import DetectorPriorityLevel
 
 
+class DataConditionType(TypedDict):
+    id: int | None
+    comparison: int
+    type: Condition
+    condition_result: DetectorPriorityLevel
+    condition_group_id: int
+
+
+class DataSourceType(TypedDict):
```

Comment:
Since this seems to be specific to snuba query's as the data source:
```suggestion
class SnubaQueryDataSourceType(TypedDict):
```